lsat/satisfier.go

```diff
@@ -115,3 +120,63 @@ func NewCapabilitiesSatisfier(service string, targetCapability string) Satisfier
 		},
 	}
 }
+
+// NewTimeoutSatisfier checks if an LSAT is expired or not.
+// The Satisfier takes a service name to set as the condition prefix
+// and currentTimestamp to compare against the expiration(s) in the caveats.
+// The expiration time is retrieved from the caveat values themselves.
+// The satisfier will also make sure that each subsequent caveat of the same
+// condition only has increasingly strict expirations.
+func NewTimeoutSatisfier(service string, currentTimestamp int64) Satisfier {
+	return Satisfier{
+		Condition: service + CondTimeoutSuffix,
+
+		SatisfyPrevious: func(prev, cur Caveat) error {
+			// check that each caveat is more restrictive than the previous
```

Comment:
that check is not happening here though but a few lines down. Also, start of sentence should start with a capital letter.